pkg/arvo/gen/validate-ship.hoon

```diff
@@ -0,0 +1,108 @@
+::  Validate userspace state
+::
+::  /hoon/validate-ship/gen
+::
+/+  resource
+:-  %say
+|=  $:  [now=@da eny=@uvJ =beak]
+        [~ ~]
+    ==
+:-  %noun
+=|  issues=(list tape)
+|^
+=.  issues
+  check-group-syncs
+issues
+::
+++  scry-synced
+  |=  app=term
+  ^-  (set path)
+  %~  key  by
+  .^  (map path ship)
+    %gx
+    (scot %p p.beak)
+    app
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  scry-grp-push-syncs
+  ^-  (set resource)
+  .^  (set resource) 
+    %gx
+    (scot %p p.beak)
+    %group-push-hook
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  scry-grp-pull-syncs
+  ^-  (set resource)
+  %~  key  by
+  .^  (map resource ship)
+    %gx
+    (scot %p p.beak)
+    %group-pull-hook
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  report-missing-syncs
+  |*  [app=term syncs=(set)]
+  ^+  issues
+  %+  weld
+    issues
+  %+  turn
+    ~(tap in syncs)
+  |=(thing=* "{<app>}-hook missing: {<thing>}")
```

Comment:
Isn't this gonna print `thing` as a plain noun? You probably want this to be a wet gate too.

```suggestion
  |*(thing=* "{<app>}-hook missing: {<thing>}")
```

Reply:
Ideally, `report-missing-syncs` wouldn't be a wet gate and would just take either a `(set path)` or a `(set resource)`, and would normalize all input to one or the other, then we wouldn't need `thing=*` at all.